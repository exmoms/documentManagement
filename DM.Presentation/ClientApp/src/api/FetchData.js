/**
 * @file this file includes functions that fetch data from API
 */

// Generic Delete Method
export async function deleteFromApi(url) {
  let res = await fetch(url, {
    method: "DELETE",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
  });
  return res;
}

// Get MetaData Models
export async function fetchData(url) {
  let object = {};
  /*
   * send the request through a CORS proxy
   * https://stackoverflow.com/questions/43262121/trying-to-use-fetch-and-pass-in-mode-no-cors/43268098#43268098
   */
  await fetch(url, {
    method: "GET",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
  })
    .then((response) => response.json())
    .then(
      (data) => (object = data),
      (error) => console.log(error)
    );
  return object;
}
