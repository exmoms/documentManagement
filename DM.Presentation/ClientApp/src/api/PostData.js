/**
 * @file This file includes functions that help to post data to a specific API
 * @version 1.0
 */

/**
 * Post new document data.
 * Upload form-data, scanned copy of doc and attachments.
 * @author Youssef Shaaban <y.shaaban@lit-co.net>
 * @param {*} documentData
 * @param {*} images
 * @param {*} attachments
 * @param {*} url
 * @param {*} option its values either "Add" for add new document version or "Update" to update exsiting document Version
 */
export async function postDocumentData(
  documentData,
  scannedPages,
  attachments,
  url = "",
  option
) {
  var formData = new FormData();

  if (process.env.NODE_ENV === "development") {
    console.log(documentData);
    console.log(attachments);
    console.log(scannedPages);
  }

  switch (option) {
    case "Add":
      formData.append("Document", JSON.stringify(documentData));
      break;
    case "Update":
      formData.append("Version", JSON.stringify(documentData));
      break;
    default:
  }

  attachments.forEach((attachment) => {
    formData.append("Attachments", attachment);
  });
  scannedPages.forEach((scannedPage) => {
    formData.append("Scans", scannedPage);
  });

  let response = {};
  await fetch(url, {
    // content-type header should not be specified!
    method: "POST",
    headers: {
      Accept: "application/json",
      // "Content-Type": "application/json",
    },
    body: formData,
  }).then(
    (res) => {
      response = res;
      // response.json();
    },
    (error) => console.log("[ERROR]:", error)
  );
  return response;
}

/**
 * Post Attachment Data.
 * Upload attachment json data + attachment file  .
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @param {*} payload
 * @param {*} attachmentFile
 * @param {*} url
 */
export async function postAttachmentData(payload, attachmentFile, url = "") {
  var formData = new FormData();

  formData.append("Attachment", JSON.stringify(payload));
  formData.append("FileAttachment", attachmentFile);

  let response = {};
  await fetch(url, {
    // content-type header should not be specified!
    method: "POST",
    body: formData,
  }).then(
    (res) => {
      response = res;
      // response.json();
    },
    (error) => console.log("[ERROR]:", error)
  );
  return response;
}

/**
 * Post Request To API
 * Send post request to api with  payload  and return a response.
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @param {*} url The api url
 * @param {*} payload js object which will be sent as request body.
 */
export async function postDataToAPI(url, payload = null, method = "POST") {

  const response = await fetch(url, {
    method: method,
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(payload),
  });
  return response;
}
