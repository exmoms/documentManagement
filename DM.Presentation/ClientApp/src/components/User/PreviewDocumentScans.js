import React, { Component } from "react";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Typography from "@material-ui/core/Typography";
import { withStyles } from "@material-ui/core/styles";
import CREATE_DOC_STYLES from "./Styles";
import { withTranslation } from "react-i18next";
const style_ = CREATE_DOC_STYLES;

class PreviewDocumentScans extends Component {
  constructor(props) {
    super(props);
    this.scannedDocInput = React.createRef();
  }
  t = this.props["t"];
  render() {
    const { classes, scannedDoc } = this.props;

    return (
      <div style={{ padding: "20px 5px" }}>
        <input
          type="file"
          id="upload-scanned-doc-input"
          required
          accept="image/*"
          ref={(ref) => (this.scannedDocInput = ref)}
          style={{ display: "none" }}
          multiple
        />
        <Card>
          <CardContent>
            <Typography gutterBottom variant="h5" component="h2" align="left">
              {this.t("previewdoc_scanneddoc")}
            </Typography>
          </CardContent>
          <React.Fragment>
            {/* @todo don't display this element until the image is selected */}
            {/*       the scanned doc may contain multi-page, so we have multi-img */}
            {/* multi-page may need a button to add new part in case user didn't use multiple selection */}
            {scannedDoc.length !== 0 && (
              <div className={classes.buttonUploadScannedDoc}>
                {scannedDoc.map((imageURI, index) => (
                  <div className={classes.docMediaContainer} key={index}>
                    <CardMedia
                      component="img"
                      image={imageURI}
                      className={classes.media}
                      id="scanned-doc-img-card"
                      title={this.t("previewdoc_scanneddoc")}
                    />
                  </div>
                ))}
              </div>
            )}
          </React.Fragment>
        </Card>
      </div>
    );
  }
}

export default withStyles(style_)(withTranslation()(PreviewDocumentScans));
