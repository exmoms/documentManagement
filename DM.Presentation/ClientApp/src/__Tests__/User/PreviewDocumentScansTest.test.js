import React from "react";
import ReactDom from "react-dom";
import PreviewDocumentScans from "../../components/User/PreviewDocumentScans";
import { cleanup, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";
import { createMount } from '@material-ui/core/test-utils';
import Dialog from "@material-ui/core/Dialog";
import CardMedia from "@material-ui/core/CardMedia";
import ReactToPrint from "react-to-print";
import * as PostDataMock from "../../api/PostData";
import { DialogContentText } from "@material-ui/core";
afterEach(cleanup);
describe("PreviewDocumentScans", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  test("renders without crashing", () => {
      let classes = {buttonUploadScannedDoc:"buttonUploadScannedDoc" , docMediaContainer: "docMediaContainer" , media:"media"};
      let scannedDoc = ["imageURI 1" , "imageURI 2"];
    const div = document.createElement("div");
    ReactDom.render(
      <PreviewDocumentScans
      classes={classes}
      scannedDoc={scannedDoc}
      />,
      div
    );
  });
  test("PreviewDocumentScans matches snapshot", () => {
    let classes = {buttonUploadScannedDoc:"buttonUploadScannedDoc" , docMediaContainer: "docMediaContainer" , media:"media"};
      let scannedDoc = ["imageURI 1" , "imageURI 2"];
    const renderedValue = createMount()(
        <PreviewDocumentScans
        classes={classes}
        scannedDoc={scannedDoc}
      />
      );
      expect(renderedValue.html()).toMatchSnapshot();
  });

  test("Form tests of Document History", () => {
    let classes = {buttonUploadScannedDoc:"buttonUploadScannedDoc" , docMediaContainer: "docMediaContainer" , media:"media"};
    let scannedDoc = ["imageURI 1" , "imageURI 2"];
    const wrapper = mount(
        <PreviewDocumentScans
        classes={classes}
        scannedDoc={scannedDoc}
      />
    );
    expect(wrapper.find(".buttonUploadScannedDoc").length).toBe(1);
    expect(wrapper.find(".docMediaContainer").length).toBe(2);
    expect(wrapper.find(CardMedia).length).toBe(2);
    expect(wrapper.find(CardMedia).at(0).props().image).toBe("imageURI 1");
    expect(wrapper.find(CardMedia).at(1).props().image).toBe("imageURI 2");
  });
  
});