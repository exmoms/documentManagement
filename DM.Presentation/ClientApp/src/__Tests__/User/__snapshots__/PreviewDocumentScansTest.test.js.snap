// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewDocumentScans PreviewDocumentScans matches snapshot 1`] = `"<div style=\\"padding: 20px 5px;\\"><input type=\\"file\\" id=\\"upload-scanned-doc-input\\" required=\\"\\" accept=\\"image/*\\" style=\\"display: none;\\" multiple=\\"\\"><div class=\\"MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded\\"><div class=\\"MuiCardContent-root\\"><h2 class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom MuiTypography-alignLeft\\">previewdoc_scanneddoc</h2></div><div class=\\"withI18nextTranslation(PreviewDocumentScans)-buttonUploadScannedDoc-15 buttonUploadScannedDoc\\"><div class=\\"withI18nextTranslation(PreviewDocumentScans)-docMediaContainer-12 docMediaContainer\\"><img class=\\"MuiCardMedia-root withI18nextTranslation(PreviewDocumentScans)-media-13 media MuiCardMedia-media MuiCardMedia-img\\" src=\\"imageURI 1\\" id=\\"scanned-doc-img-card\\" title=\\"previewdoc_scanneddoc\\"></div><div class=\\"withI18nextTranslation(PreviewDocumentScans)-docMediaContainer-12 docMediaContainer\\"><img class=\\"MuiCardMedia-root withI18nextTranslation(PreviewDocumentScans)-media-13 media MuiCardMedia-media MuiCardMedia-img\\" src=\\"imageURI 2\\" id=\\"scanned-doc-img-card\\" title=\\"previewdoc_scanneddoc\\"></div></div></div></div>"`;
